pkg/controllers/metrics/node/counter/counter.go

```diff
@@ -0,0 +1,166 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package counter
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	karpenterapi "github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha4"
+	"github.com/awslabs/karpenter/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	metricArchLabel                    = "arch"
+	metricConditionDiskPressureLabel   = "diskpressure"
+	metricConditionMemoryPressureLabel = "memorypressure"
+	metricConditionPIDPressureLabel    = "pidpressure"
+	metricConditionReadyLabel          = "ready"
+	metricInstanceTypeLabel            = "instancetype"
+	metricTopologyRegionLabel          = "region"
+	metricTopologyZoneLabel            = "zone"
+	metricOsLabel                      = "os"
+)
+
+var (
+	nodeCountGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metrics.KarpenterNamespace,
+			Subsystem: "cluster",
```

Comment:
Thoughts on calling this capacity? Conceptually
- "Karpenter Capacity Metrics" = "How good of a job is Karpenter doing managing my capacity? 
- "Karpenter Controller Metrics" = "How healthy is the karpenter process itself?"